download_artifacts.js

```diff
@@ -0,0 +1,416 @@
+const fs = require('fs');
+const axios = require('axios');
+const stream = require('stream');
+const util = require('util');
+const { program } = require('commander');
+const commander = require('commander');
+const yesno = require('yesno');
+
+const finished = util.promisify(stream.finished);
+
+// -------------------- CONSTANTS --------------------
+
+const GITHUB_API_BASEURL = 'https://api.github.com';
+const NEUTRON_ORG = 'neutron-org';
+const STORAGE_ADDR_BASE =
+  'https://storage.googleapis.com/neutron-contracts/neutron-org';
+const DEFAULT_BRANCH = 'neutron_audit_informal_17_01_2023';
+const DEFAULT_DIR = 'contracts';
+const CI_TOKEN_ENV_NAME = 'PAT_TOKEN';
+const DEFAULT_TIMEOUT = 15 * 60;
+const DELAY_BETWEEN_TRIES = 10;
+const WORKFLOW_YAML_NAME = 'build.yml';
+
+// -------------------- GLOBAL_VARS --------------------
+
+let REWRITE_FILES = false;
+
+// -------------------- UTILS --------------------
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+let verboseLog = () => {};
+
+const downloadFile = async (fileUrl, outputLocationPath) => {
+  verboseLog(`Downloading file by url: ${fileUrl}`);
+  const writer = fs.createWriteStream(outputLocationPath);
+  return axios({
+    method: 'get',
+    url: fileUrl,
+    responseType: 'stream',
+  }).then((response) => {
+    response.data.pipe(writer);
+    return finished(writer);
+  });
+};
+
+const wait = async (seconds) =>
+  new Promise((r) => {
+    setTimeout(() => r(true), 1000 * seconds);
+  });
+
+const getWithAttempts = async (getFunc, readyFunc, numAttempts = 20) => {
+  let error = null;
+  let data = null;
+  const delay = DELAY_BETWEEN_TRIES;
+  while (numAttempts > 0) {
+    numAttempts--;
+    try {
+      data = await getFunc();
+      if (await readyFunc(data)) {
+        return data;
+      }
+    } catch (e) {
+      error = e;
+    }
+    console.log(`${numAttempts * delay} seconds left`);
+    await wait(delay);
+  }
+  throw error != null
+    ? error
+    : new Error(
+        'getWithAttempts: no attempts left. Latest get response: ' +
+          (data === Object(data) ? JSON.stringify(data) : data).toString(),
+      );
+};
+
+const askForRewrite = async (file_name) => {
+  const ok = await yesno({
+    question: `File ${file_name} already exists, do you want to rewrite? (if yes, all files are to be rewritten)`,
```

Comment:
```suggestion
    question: `File ${file_name} already exists, do you want to overwrite it? (if yes, all further differing files will be overwritten)`,
```

I believe this version would be more concise.

Reply:
Agree